src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterSelect } from './types/FilterSelect';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = React.useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<string>('');
+
+  const [selectedTodo, setSelectedTodo] = React.useState<Todo | null>(null);
+
+  const [filterSelect, setFilterSelect] = React.useState<FilterSelect>(FilterSelect.All);
+  const [filterTitle, setFilterTitle] = React.useState<string>('');
+
+  const filterTodos = (
+    todosToFilter: Todo[],
+    filterToApply: [FilterSelect, string],
+  ) => {
+    let newTodos = [...todosToFilter];
+
+    newTodos = newTodos.filter((todo) => {
+      switch (filterToApply[0]) {
+        case FilterSelect.Active:
+          return !todo.completed;
+
+        case FilterSelect.Completed:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+
+    newTodos = newTodos.filter((todo) => todo.title.trim().toLowerCase()
+      .includes(filterToApply[1].trim().toLocaleLowerCase()));
+
+    return newTodos;
+  };
```

Comment:
A bit hard for understanding `filterToApply[1] or filterToApply[0]`... might be better to use an `object` with filters instead of `array` of filterToApplywhat. it could be difficult in future if you need to add more filtering options
for instance ..
```
const filterTodos = (
  todos: Todo[],

  {
    filterSelect: status,
    filterTitle: query,
  }: { filterSelect: FilterSelect; filterTitle: string },

) => {
  let newTodos = [...todos];

  newTodos = newTodos.filter((todo) => {
    switch (status) {
      case FilterSelect.Active:
        return !todo.completed;

      case FilterSelect.Completed:
        return todo.completed;

      default:
        return true;
    }
  });

  if (query) {
    const normalizedQuery = query.trim().toLowerCase();

    newTodos = newTodos.filter((todo) => {
      const normalizedTodoTitle = todo.title.trim().toLowerCase();

      return normalizedTodoTitle.includes(normalizedQuery);
    });
  }

  return newTodos;
};
```